File: 3223/go/main.go
package main

import "fmt"

func minimumLength(s string) int {
	if len(s) <= 2 {
		return len(s)
	}

	counter := make(map[rune]int)

	for _, r := range s {
		if _, ok := counter[r]; !ok {
			counter[r] = 1
			continue
		}

		counter[r] += 1
	}

	min_size := len(s)

	for _, freq := range counter {
		if freq <= 2 {
			continue
		}
		if freq%2 == 0 {
			min_size -= freq - 2
			continue
		}

		min_size -= freq - 1

	}

	return min_size
}

func main() {
	fmt.Println(minimumLength("abaacbcbb"))
	fmt.Println(minimumLength("aa"))
}

File: 3223/python/main.py
from collections import Counter


class Solution:
    def minimumLength(self, s: str) -> int:
        if len(s) <= 2:
            return len(s)
        c = Counter(s)

        min_size = len(s)

        for _, freq in c.items():
            if freq <= 2:
                continue

            # f - x = 2; f-2 = x

            if freq % 2 == 0:
                min_size -= freq - 2
                continue

            min_size -= freq - 1

        return min_size


s = Solution()

print(s.minimumLength("abaacbcbb"))
print(s.minimumLength("aa"))
